### PR TITLE
CMake improvements

### PR DESCRIPTION
**Pull request by compnerd**
This is a collection of various improvements to use more modern CMake constructs.  This makes the build less reliant on the specifics of the compiler that you are using and just simplifies build rules.

### CODE REVIEW

File: cmake/modules/FindSQLite3.cmake [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Distributed under the OSI-approved BSD 3-Clause License.  See accompanying
```
Comment (ddunbar) [inline cmake/modules/FindSQLite3.cmake]: Please add a reference to this from LICENSE.txt (see the bottom)

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
Ah right, need to adjust the swift build side ..

**Comment by compnerd** [comment]
*sigh* typos ...

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Pull request merged**